File: QueueImplementation/Queue.py

class Queue:
    Q=[]
    def enque(self,ele):
        self.Q.append(ele)
    def deque(self):
        if(len(self.Q)>0):
            ele=self.Q[0]
            self.Q=self.Q[1:]
            return ele
        return -1
    def printall(self):
        if(len(self.Q)>0):
            print("The elements in Queue are: ",end=" ")
            for i in self.Q:
                print(i,end=" ")
            print()
        else:
            print("Queue is empty")
    def printFront(self):
        if(len(self.Q)>0):
            print("The Front of the Queue is",self.Q[0])
        else:
            print("Queue is Empty")
    def printRear(self):
        if(len(self.Q)>0):
            print("The Rear of the Queue is",self.Q[-1])
        else:
            print("Queue is Empty")
    
class Main:
   def main():
        qObj=Queue()
        while(True):
            print("1.enqueue   2.dequeue   3.PrintQueue   4.printFront   5.printRear   6.Exit")
            choice=int(input("enter your choice: "))
            if choice==1:
                ele=input("enter the element: ")
                qObj.enque(ele)
                print("The element",ele,"is successfully inserted")
            elif choice==2:
                ele=qObj.deque()
                if ele!=-1:
                    print("The element",ele,"is successfully deleted")
                    continue
                print("The Queue is Empty")
            elif choice==3:
                qObj.printall()
            elif choice==4:
                qObj.printFront()
            elif choice==5:
                qObj.printRear()
            else:
                print("Exiting Program")
                break
   if __name__=="__main__":
         main()
    
    

File: QueueImplementation/Queue.java
import java.util.Scanner;


class QueueClass
{
    int[] arr;
    int size,front=0,rear=0;
     public QueueClass(int arrsize)
     {
        size=arrsize;
       arr=new int[size];
     }
     public void enqueu(int ele)
     {
        arr[rear]=ele;
        rear=(rear+1)%size;
        System.out.println("Element "+ele+" is successfully inserted");
     }
     public void deque()
     {   int ele;
         ele=arr[front];
         front=(front+1)%size;
         System.out.println("Element "+ele+" is dequed from queue");
        
     }
     public void printAll()
     {
         System.out.print("The elements in queue are: ");
         int i=front;
         int j=rear;
         while(i!=j)
         {
            System.out.print(arr[i]+" ");
            i=(i+1)%5;
         }
           
      
        System.out.println();
     }

     public void printFront()
     {
         System.out.println("The front of the queue is "+arr[front]+" and front index is "+front);
     }
     public void printRear()
     {
         int re=(rear-1)%5;
         System.out.println("The rear of the queue is "+arr[re]+" and rear index is "+re);
     }

}


class Queue 
{
   public static void main(String[] args) {
       Scanner sc=new Scanner(System.in);
       int size,choice,ele;
       System.out.print("Enter the size of queue: ");
       size=sc.nextInt();
       QueueClass qObj=new QueueClass(size);
       while(true)
       {
           System.out.println("1.Enqueue   2.Dequeue   3.PrintAll   4.PrintFront   5.PrintRear   6.Exit");
           System.out.print("Enter your choice: ");
           choice=sc.nextInt();
           switch(choice)
           {
                case 1:
                        if((qObj.rear+1)%size==qObj.front)
                        {
                            System.out.println("OverFlow: Queue is Full");
                            break;
                        }
                        System.out.print("Enter the element to enqueue: ");
                        ele=sc.nextInt();
                        qObj.enqueu(ele);
                        break;   
                case 2:
                      if(qObj.front==qObj.rear)
                      {
                        System.out.println("UnderFlow: Queue is Empty");
                        break;
                      }
                      qObj.deque();
                      break;
                case 3:
                        if(qObj.front==qObj.rear)
                        {
                        System.out.println("Queue is Empty");
                        break;
                        }
                        qObj.printAll();
                        break;
                case 4:
                    if(qObj.front==qObj.rear)
                            {
                            System.out.println("Queue is Empty");
                            break;
                            }
                            qObj.printFront();
                            break;
                case 5:
                        if(qObj.front==qObj.rear)
                        {
                        System.out.println("Queue is Empty");
                        break;
                        }
                        qObj.printRear();
                        break;
                 default:
                 System.out.println("Exiting Program....");
                 break;
            }
            if(choice>5)
            break;
       }
       sc.close();

   }
    
}
